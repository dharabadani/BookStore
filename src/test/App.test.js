import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import { store, history } from '../configureStore';
import routes from '../routes';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import createRootReducer from '../rootReducer'

jest.mock('../configureStore', () => {
    return {
        store: {subscribe: jest.fn(), dispatch: jest.fn(), getState: jest.fn()},
        history: {listen: jest.fn(), action: '', location: {}, push: jest.fn()}
    }
});
jest.mock('../rootReducer')
jest.mock('../routes', () => {
    return (
        <div></div>
    );
});

test('Test App', () => {
    const { store } = require('../configureStore');
    const { history } = require('../configureStore');
    const app = renderer.create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
});