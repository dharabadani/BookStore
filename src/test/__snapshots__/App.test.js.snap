// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App 1`] = `
<div
  className="app"
>
  <header
    data-testid="app-header"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="#/"
      >
        Book Store
      </a>
    </nav>
  </header>
  <main>
    <div />
  </main>
  <footer
    data-testid="app-footer"
  >
    Â© 
    2021
  </footer>
</div>
`;
